### ISSUE TITLE
Ptycho.run crashes for empty scans - streaming breaks

### ISSUE THREAD

**Issue by alexbjorling** [ptycho/ptypy#237]
I am playing with a simple `PtyScan` class which takes a zmq stream from my data acquisition code and reconstructs it live (see https://github.com/ptycho/ptypy/blob/nanomax_zmq/ptypy/experiment/contrast.py) Problem is that even if `check()` returns 0 (that is, the scan hasn't started yet), the main loop in `Ptycho.run()` goes on to run the engines after calling `new_data()`. Even when working around the resulting division-by-zero in `DM`, it crashes because the returned errors make no sense.

Right now my `check()` blocks execution as a workaround until enough frames are collected, but I feel that `Ptycho`should tolerate empty scans to allow streaming data. What do you think?

**Comment by bjoernenders** [comment]
Yes we need to make the new_data call a bit smarter to deal with asynchronous data acquisition. I think I already started with that on the `faster_scans` branch, but let me check. I'm just back from vacation.

**Comment by bjoernenders** [comment]
relates to #177

**Comment by bjoernenders** [comment]
live processing needs to be addressed in the next release

**Comment by daurer** [comment]
relates to #466
